src/pages/MainPage.js

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import { Link } from 'react-router-dom';
+
+const MainPage = () => {
+  const [resorts, setResorts] = useState([]);
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  useEffect(() => {
+    // Fetch the resorts data from the backend
+    const fetchResorts = async () => {
+      try {
+        const response = await axios.get('http://localhost:4000/resorts');
+        setResorts(response.data);
+      } catch (error) {
+        console.error('Error fetching resorts:', error);
+      }
```

Comment:
**OPTIONAL**
Before submitting your code, it would be beneficial to remove the console statement to prevent any potential linter errors. Feel free to research different error-handling approaches to find the one that best suits your specific use case☺☺